haystack/components/generators/chat/openai.py

```diff
@@ -284,13 +284,12 @@ def run(
         else:
```

Comment:
@vblagoje all changes in this file (the one containing the ChatGenerator) have to do with making methods not class functions anymore. So it looks like a large diff, but it contains no actual changes other than dropping `self`